examples/typescript/keyless.ts

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable max-len */
+/* eslint-disable no-console */
+
+/**
+ * This example shows how to use the Keyless accounts on Aptos
+ */
+
+import { Account, AccountAddress, Aptos, AptosConfig, EphemeralKeyPair, Network } from "@aptos-labs/ts-sdk";
+import * as readlineSync from "readline-sync";
+
+const COIN_STORE = "0x1::coin::CoinStore<0x1::aptos_coin::AptosCoin>";
+const ALICE_INITIAL_BALANCE = 100_000_000;
+const BOB_INITIAL_BALANCE = 100;
+const TRANSFER_AMOUNT = 10_000;
+
+/**
+ * Prints the balance of an account
+ * @param aptos
+ * @param name
+ * @param address
+ * @returns {Promise<*>}
+ *
+ */
+const balance = async (aptos: Aptos, name: string, address: AccountAddress) => {
+  type Coin = { coin: { value: string } };
+  const resource = await aptos.getAccountResource<Coin>({
+    accountAddress: address,
+    resourceType: COIN_STORE,
+  });
+  const amount = Number(resource.coin.value);
+
+  console.log(`${name}'s balance is: ${amount}`);
+  return amount;
+};
+
+const example = async () => {
+  // Setup the client
+  const config = new AptosConfig({ network: Network.DEVNET });
+  const aptos = new Aptos(config);
+
+  // Generate the ephemeral (temporary) key pair that will be used to sign transactions.
+  const aliceEphem = EphemeralKeyPair.generate();
+
+  console.log("\n=== Keyless Account Example ===\n");
+
+  const link = `https://accounts.google.com/o/oauth2/v2/auth/oauthchooseaccount?redirect_uri=https%3A%2F%2Fdevelopers.google.com%2Foauthplayground&prompt=consent&response_type=code&client_id=407408718192.apps.googleusercontent.com&scope=openid&access_type=offline&service=lso&o2v=2&theme=glif&flowName=GeneralOAuthFlow&nonce=${aliceEphem.nonce}`;
+  console.log(`${link}\n`);
+
+  console.log("1. Open the link above");
+  console.log("2. Log in with your Google account");
+  console.log("3. Click 'Exchange authorization code for tokens");
+  console.log("4. Copy the 'id_token' - (toggling 'Wrap lines' option at the bottom makes this easier)\n");
+
+  function inputJwt(): string {
+    const jwt: string = readlineSync.question("Paste the JWT (id_token) token here and press enter: ", {
+      hideEchoBack: false,
+    });
+    return jwt;
+  }
+
+  const jwt = inputJwt();
+  // Derive the Keyless Account from the JWT and ephemeral key pair.
+  const alice = await aptos.deriveKeylessAccount({
+    jwt,
+    ephemeralKeyPair: aliceEphem,
+  });
+
+  console.log("=== Addresses ===\n");
+  console.log(`Alice's keyless account address is: ${alice.accountAddress}`);
+  console.log(`Alice's nonce is: ${aliceEphem.nonce}`);
+  console.log(`Alice's ephem pubkey is: ${aliceEphem.getPublicKey().toString()}`);
+
+  const bob = Account.generate();
+  console.log(`Bob's address is: ${bob.accountAddress}`);
+
+  // Fund the accounts
+  console.log("\n=== Funding accounts ===\n");
+
+  await aptos.faucet.fundAccount({
+    accountAddress: alice.accountAddress,
+    amount: ALICE_INITIAL_BALANCE,
+    options: { waitForIndexer: false },
```

Comment:
why we dont wait for indexer?

Reply:
ah i added this when testing locally which i usually don't run with indexer.  but for devnet i can remove.  done